File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/utils/ThreadManager.java
package me.ycdev.android.lib.common.utils;

import android.os.HandlerThread;
import android.os.Looper;

public class ThreadManager {
    private static ThreadManager sInstance = new ThreadManager();

    private HandlerThread mLocalServiceRequestIpcThread;
    private HandlerThread mRemoteServiceRequestIpcThread;

    public static ThreadManager getInstance() {
        return sInstance;
    }

    public synchronized Looper localServiceRequestIpcLooper() {
        if (mLocalServiceRequestIpcThread == null) {
            mLocalServiceRequestIpcThread = new HandlerThread("LocalService.client");
            mLocalServiceRequestIpcThread.start();
        }
        return mLocalServiceRequestIpcThread.getLooper();
    }

    public synchronized Looper remoteServiceRequestIpcLooper() {
        if (mRemoteServiceRequestIpcThread == null) {
            mRemoteServiceRequestIpcThread = new HandlerThread("RemoteService.client");
            mRemoteServiceRequestIpcThread.start();
        }
        return mRemoteServiceRequestIpcThread.getLooper();
    }
}

File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/utils/ThreadManager.kt
package me.ycdev.android.lib.common.utils

import android.os.HandlerThread
import android.os.Looper

class ThreadManager {

    private var mLocalServiceRequestIpcThread: HandlerThread? = null
    private var mRemoteServiceRequestIpcThread: HandlerThread? = null

    @Synchronized
    fun localServiceRequestIpcLooper(): Looper {
        if (mLocalServiceRequestIpcThread == null) {
            mLocalServiceRequestIpcThread = HandlerThread("LocalService.client")
            mLocalServiceRequestIpcThread!!.start()
        }
        return mLocalServiceRequestIpcThread!!.looper
    }

    @Synchronized
    fun remoteServiceRequestIpcLooper(): Looper {
        if (mRemoteServiceRequestIpcThread == null) {
            mRemoteServiceRequestIpcThread = HandlerThread("RemoteService.client")
            mRemoteServiceRequestIpcThread!!.start()
        }
        return mRemoteServiceRequestIpcThread!!.looper
    }

    companion object {
        val instance = ThreadManager()
    }
}
